modules/kernel/src/main/scala/org/tessellation/kernel/StateChannelDef.scala

```diff
@@ -1,16 +1,21 @@
 package org.tessellation.kernel
 
-import cats.Monad
+import cats.effect.Async
 
 import org.tessellation.schema.address.Address
 
+import fs2.Pipe
+
 trait StateChannelDef[A <: Ω, B <: Ω, S <: Ω] {
 
   val address: Address
   val kryoRegistrar: Map[Class[_], KryoRegistrationId]
 
-  def makeCell[F[_]: Monad](
+  def makeCell[F[_]: Async](
     input: A,
     hgContext: HypergraphContext[F]
   ): Cell[F, StackF, A, Either[CellError, B], S]
+
+  def inputPipe[F[_]: Async, In <: A]: Pipe[F, In, A] = identity
+  def outputPipe[F[_]: Async, Out >: B]: Pipe[F, B, Out] = identity
```

Comment:
Why are these type constraints `In <: A` and `Out >: B` necessary?

Reply:
Should it be A => A and B => B ?

Reply:
No, I would leave it as it is but without any constraints, unless they're necessary. `In => A` and `B => Out`.